lib/cli/run.js

```diff
@@ -0,0 +1,403 @@
+        default: './test/mocha.opts',
+        description: 'Path to `mocha.opts`',
+        group: GROUPS.CONFIG,
+        normalize: true,
+        requiresArg: true
+      },
+      recursive: {
+        description: 'Look for tests in subdirectories',
+        group: GROUPS.FILES,
+        type: 'boolean'
+      },
+      reporter: {
+        alias: 'R',
+        default: 'spec',
+        description: 'Specify reporter to use',
+        group: GROUPS.OUTPUT,
+        requiresArg: true,
+        type: 'string'
+      },
+      reporters: {
+        description: 'List built-in reporters & exit',
+        type: 'boolean'
+      },
+      require: {
+        alias: 'r',
+        default: [],
+        defaultDescription: '(none)',
+        description: 'Require module',
+        group: GROUPS.FILES,
+        requiresArg: true,
+        type: 'array'
+      },
+      retries: {
+        defaultDescription: '0',
+        description: 'Retry failed tests this many times',
+        group: GROUPS.RULES,
+        type: 'number'
+      },
+      slow: {
+        alias: 's',
+        default: 75,
+        description: 'Specify "slow" test threshold (in milliseconds)',
+        group: GROUPS.RULES,
+        type: 'number'
+      },
+      sort: {
+        alias: 'S',
+        description: 'Sort test files',
+        group: GROUPS.FILES,
+        type: 'boolean'
+      },
+      timeout: {
+        alias: 't',
+        default: 2000,
+        description: 'Specify test timeout threshold (in milliseconds)',
+        group: GROUPS.RULES
+      },
+      ui: {
+        alias: 'u',
+        default: 'bdd',
+        description: 'Specify user interface',
+        group: GROUPS.RULES,
+        requiresArg: true,
+        type: 'string'
+      },
+      watch: {
+        alias: 'w',
+        description: 'Watch files in the current working directory for changes',
+        group: GROUPS.FILES,
+        type: 'boolean'
+      }
+    })
+    .positional('spec', {
+      default: ['test/'],
+      defaultDescription: 'test',
+      description: 'One or more files, directories, or globs to test',
+      type: 'array'
+    })
+    .config('config', 'Load config file', filepath => {
+      // yargs normalizes this into an absolute path, which is why
+      // this is weird.
+      if (filepath === path.resolve(CONFIG_PATH_PLACEHOLDER)) {
+        debug(`looking for config from ${cwd}`);
+        return findConfig(cwd);
+      }
+      debug(`loading config at ${filepath}`);
+      return loadConfig(filepath);
+    })
+    .pkgConf('mocha')
+    .check(argv => {
+      if (argv.invert && !('fgrep' in argv || 'grep' in argv)) {
+        throw new Error(
+          '"--invert" requires one of "--fgrep <str>" or "--grep <regexp>"'
+        );
+      }
+
+      validatePlugin(argv, 'reporter', Mocha.reporters);
+      validatePlugin(argv, 'ui', Mocha.interfaces);
+
+      return true;
+    });
+
+exports.handler = argv => {
+  debug('post-yargs config', argv);
+
```

Comment:
One'n'Dones can be handled within `yargs`. See [here](https://github.com/cybertk/abao/blob/eb850ef47cbe14964c5951746f5485109cd57caf/lib/cli.coffee#L32-L60).

Reply:
I don't understand.

Reply:
one'n'done: cmdline option treated like `--version`. If option present, execute its function and terminate. For Mocha, `--reporters` and `--interfaces`.

Reply:
ok, so in `check()`, essentially.  

- [ ] do this